drivers/sensor/iis2iclx/iis2iclx.c

```diff
@@ -242,10 +251,20 @@ static int iis2iclx_sample_fetch_shub(const struct device *dev)
 static int iis2iclx_sample_fetch(const struct device *dev,
 				   enum sensor_channel chan)
 {
+#if defined(CONFIG_IIS2ICLX_SENSORHUB)
+	struct iis2iclx_data *data = dev->data;
+#endif
+
 	switch (chan) {
 	case SENSOR_CHAN_ACCEL_XYZ:
 		iis2iclx_sample_fetch_accel(dev);
+
 #if defined(CONFIG_IIS2ICLX_SENSORHUB)
+		if (!data->shub_inited) {
```

Comment:
Is it not possible to pull this check before the switch/case statement?

Reply:
Actually no. If shub_inited is false, means that there are no slave devices attached to it, so you need to avoid to call iis2iclx_sample_fetch_shub(dev), but still you need to call iis2iclx_sample_fetch_accel(dev).

What I can do instead is to move the shub_inited check inside the iis2iclx_sample_fetch_shub() itself, so that I could avoid all this checks. Does it make sense to you?

Reply:
> What I can do instead is to move the shub_inited check inside the iis2iclx_sample_fetch_shub() itself, so that I could avoid all this checks. Does it make sense to you?

that make sense.

Reply:
ok, then

Reply:
did it

Reply:
@galak 
Can you check this point again? 
I think now should be ok